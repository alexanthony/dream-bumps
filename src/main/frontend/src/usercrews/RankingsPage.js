// @flow
import React, {Component, PropTypes} from 'react';
import { connect } from 'react-redux';
import {Tabs, Tab} from 'material-ui/Tabs';

import RankingGenderTabContent from './RankingGenderTabContent';
import { fetchUserCrewRankings } from './actions';
import { fetchUsers } from '../admin/actions';

class RankingsPage extends Component {
  componentDidMount() {
    this.props.fetchUserCrewRankings('male');
    this.props.fetchUserCrewRankings('female');
    this.props.fetchUsers();
  }

  render() {
    return (
      <Tabs>
        <Tab label="Combined">
          <RankingGenderTabContent sex="combined" />
        </Tab>
        <Tab label="Men">
          <RankingGenderTabContent sex="male" />
        </Tab>
        <Tab label="Women">
          <RankingGenderTabContent sex="female" />
        </Tab>
      </Tabs>
    );
  }
}

RankingsPage.propTypes = {
  fetchUsers: PropTypes.func,
  fetchUserCrewRankings: PropTypes.func
}

export {RankingsPage};

export default connect(null, {fetchUserCrewRankings, fetchUsers})(RankingsPage);
