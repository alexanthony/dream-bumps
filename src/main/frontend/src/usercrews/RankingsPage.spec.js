import React from 'react';
import { shallow } from 'enzyme';
import { RankingsPage } from './RankingsPage';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const fetchUsers = jest.fn();
  const fetchUserCrewRankings = jest.fn();
  const component = shallow(
    <RankingsPage
      fetchUsers={fetchUsers}
      fetchUserCrewRankings={fetchUserCrewRankings}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
  // We don't actually mount, so these don't get called...
  // expect(fetchUsers).toHaveBeenCalled();
  // expect(fetchUserCrewRankings).toHaveBeenCalled()
});
