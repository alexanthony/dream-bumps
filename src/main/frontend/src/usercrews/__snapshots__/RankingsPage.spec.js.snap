exports[`test Renders properly 1`] = `
<Tabs
  initialSelectedIndex={0}
  onChange={[Function]}>
  <Tab
    label="Combined">
    <Connect(RankingGenderTabContent)
      sex="combined" />
  </Tab>
  <Tab
    label="Men">
    <Connect(RankingGenderTabContent)
      sex="male" />
  </Tab>
  <Tab
    label="Women">
    <Connect(RankingGenderTabContent)
      sex="female" />
  </Tab>
</Tabs>
`;
